### ISSUE TITLE
[Bug]: Java Exception throws in Ballerina regex code

### ISSUE THREAD

**Issue by SasinduDilshara** [ballerina-platform/ballerina-lang#39448]
### Description

```
import ballerina/io;

public function main() {
  string:RegExp x7 = re `([(a{1})-(z)])`;
  io:println(x7.findAll(":A{1,2}})").toBalString());
}
```

Gives

```
Compiling source
        test.bal

Running executable

error: java.util.regex.PatternSyntaxException {"message":"Illegal character range near index 9
([(a{1})-(z)])
         ^"}
        at ballerina.lang.regexp.0:findAllImpl(regexp.bal:86)
           ballerina.lang.regexp.0:findAll(regexp.bal:136)
           test:main(test.bal:5)

```

### Related area

-> Compilation

**Comment by SasinduDilshara** [comment]
Same behaviour for

```
import ballerina/io;

public function main() {
  string:RegExp x7 = re `(?i-s:[[A\\sB\WC\Dd\\]\])`;
  io:println(x7.findAll(":*1*7").toBalString());
}

```

**Comment by SasinduDilshara** [comment]
Same Behaviour for

```
import ballerina/io;

public function main() {
  string:RegExp x7 = re `(?xsmi:[]\P{sc=Braille})`;
  io:println(x7.findAll(":*A*a").toBalString());                                  
}
```

**Comment by sanjana** [comment]
Fixed in #39444.
